packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+      style: style,
+      emptyDescription: emptyDescription,
+    );
+  }
+
+  /// Label describing the Diagnostics node.
+  ///
+  /// * If the DiagnosticsNode represents property, the [name] indicates the
+  ///   name of the property.
+  /// * If the DiagnosticsNode is a child of another DiagnosticsNode, the name
+  ///   describes the relationship between the parent and the child. For
+  ///   example: `child 5`.
+  final String name;
+
+  /// Header with a short text description of the node itself not including
+  /// children or properties.
+  final String header;
+
+  /// Whether to show if the value is null.
+  final bool showNull;
+
+  /// Whether the node is explicitly hidden.
+  final bool _hidden;
+
+  /// Whether to show a separator between [name] and [header].
+  ///
+  /// If false, name and header should be shown with no separation.
+  /// `:` is typically used as a separator when displaying as text.
+  final bool showSeparator;
+
+  /// Optional defaultValue the object has.
+  /// If [object] equals [defaultValue] and [defaultValue] is not null then
+  /// the node is [hidden].
+  final Object defaultValue;
+
+  /// Whether the diagnostics should be hidden when showing the default
+  /// view of a tree. Some displays of a tree could provide an affordance to
+  /// display hidden diagnostics.
+  bool get hidden {
+    if (_hidden)
+      return true;
+
+    if (defaultValue != null) {
+      try {
+        if (object == defaultValue)
+          return true;
+      } catch (e) {
+        // An object that throws an exception can't be null so the property
+        // should be displayed regardless.
+        return false;
+      }
+    }
+
+    if (showNull)
+      return false;
+
+    try {
+      return object == null;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /// Whether to show the name.
+  final bool showName;
+
+  /// Description to show if the node has no displayed properties or children.
+  final String emptyDescription;
+
+  /// Dart object this is diagnostics data for.
+  ///
+  /// May throw an exception.
+  Object get object;
+
+  /// Style this node should be displayed using.
+  final DiagnosticsTreeStyle style;
+
+  /// Properties of this DiagnosticsNode.
+  ///
+  /// Properties and children are kept distinct even though they are both
+  /// [List<DiagnosticNode>] because they tend to be displayed differently as
+  /// properties and in some use cases gettig
+  List<DiagnosticsNode> getProperties();
+
+  /// Children of this DiagnosticsNode.
+  List<DiagnosticsNode> getChildren();
+
+  String get _separator => showSeparator ? ':' : '';
+
+  @override
+  String toString() {
+    if (style == DiagnosticsTreeStyle.singleLine)
+      return toStringDeep();
+
+    if (name == null || name.isEmpty || showName == false)
+      return header;
+
+    return header.contains('\n') ?
+        '$name$_separator\n$header' : '$name$_separator $header';
+  }
+
+  _TextTreeConfiguration get _textTreeConfiguration {
```

Comment:
Why is this private? It seems like it would prevent someone from subclassing this class and providing different styles.